src/plugins/tools/apipTextToSpeech/plugin.js

```diff
@@ -0,0 +1,238 @@
+                } else {
+                    enablePlugin();
+                }
+            }
+        };
+
+        // Add plugin button to toolbox
+        this.button = this.getAreaBroker()
+            .getToolbox()
+            .createEntry({
+                className: `${this.getName()}-plugin`,
+                control: this.getName(),
+                icon: 'headphones',
+                text: __('Text To Speech'),
+                title: __('Enable text to speech'),
+            });
+
+        // Handle plugin button click
+        this.button.on('click', (e) => {
+            e.preventDefault();
+            testRunner.trigger(`${actionPrefix}toggle`);
+        });
+
+        // Register plugin shortcuts
+        if (testRunnerOptions.allowShortcuts) {
+            _.forEach(pluginShortcuts, (command, key) => {
+                shortcut.add(
+                    namespaceHelper.namespaceAll(command, pluginName, true),
+                    () => {
+                        const eventKey = key.endsWith('TogglePlayback') ? 'togglePlayback' : key;
+
+                        testRunner.trigger(actionPrefix + eventKey);
+                    },
+                    {
+                        avoidInput: true
+                    }
+                );
+            });
+        }
+
+        //start disabled
+        togglePlugin();
+        this.disable();
+
+        //update plugin state based on changes
+        testRunner
+            .on('loaditem', () => {
+                togglePlugin();
+                this.disable();
+            })
+            .on('enabletools renderitem', () => {
+                this.enable();
+            })
+            .on('disabletools unloaditem', () => {
+                this.disable();
+            })
+            .on(`${actionPrefix}toggle`, () => {
+                if (isConfigured()) {
+                    toggleTool();
+                }
+            })
+            .on(`${actionPrefix}togglePlayback`, () => {
+            })
+            .on(`${actionPrefix}next`, () => {
+                if (this.getState('enabled')) {
+                    if (this.getState('active')) {
+                        // handling goes here
+                    }
+                }
+            })
+            .on(`${actionPrefix}previous`, () => {
+                if (this.getState('enabled')) {
+                    if (this.getState('active')) {
+                        // handling goes here
+                    }
+                }
+            })
+            .on('renderitem', () => {
+                const $navigationGroupElement = this.button.getElement();
+                const groupNavigationId = `${pluginName}_navigation_group`;
+
+                if (!$navigationGroupElement) {
+                    return;
+                }
+
+                this.navigationGroup = keyNavigator({
+                    id: groupNavigationId,
+                    elements: navigableDomElement.createFromDoms($navigationGroupElement),
```

Comment:
I'm curious about the elements you need to navigate through. Are they the menus or are they the audio elements?
Here it look like they are the menus since they come from the tool button.

Reply:
User will be able to navigate between content parts with associated media files. From the technical requirements:
the user can use their keyboard to focus on the sequence of access elements (display the hover effect on focus) by tabbing through the content chunks (shift-TAB to go backwards in the sequence). Users may initiate playback from the current focused content chunk by pressing ENTER or SPACEBAR.
The requirements can be found here: https://oat-sa.atlassian.net/wiki/spaces/UX/pages/118522064/TAO+9016+-+SPIKE+-+Story+breakout+for+TTS+Text+to+Speech
Also this will relay on APIP test item data. The APIP data includes the order for the content parts and we will need to navigate between them according to this order. The idea is to listen to tab navigation events and change items dynamically according to the order, but it will be solved during implementation of the component for this plugin

Reply:
ok, got it